process explorer/backend/LocalCollector/EnvironmentVariables/EnvironmentVariableMonitor.cs

```diff
@@ -0,0 +1,31 @@
+﻿/* Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License"). You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0. See the NOTICE file distributed with this work for additional information regarding copyright ownership. Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License. */
+
+using System.Collections;
+using System.Collections.Concurrent;
+
+namespace ProcessExplorer.Entities.EnvironmentVariables
+{
+    public class EnvironmentMonitorDto
+    {
+        public ConcurrentDictionary<string, string>? EnvironmentVariables { get; set; } = new ConcurrentDictionary<string, string>();
+
+        public static EnvironmentMonitorDto FromEnvironment()
+        {
+            var envs = new EnvironmentMonitorDto();
+            if (envs.EnvironmentVariables != null)
+            {
+                lock (envs.EnvironmentVariables)
```

Comment:
Guidelines
When you synchronize thread access to a shared resource, lock on a dedicated object instance (for example, private readonly object balanceLock = new object();) or another instance that is unlikely to be used as a lock object by unrelated parts of the code. 
https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/statements/lock